substrate/client/executor/polkavm/src/lib.rs

```diff
@@ -43,38 +44,18 @@ impl WasmInstance for Instance {
 		name: &str,
 		raw_data: &[u8],
 	) -> (Result<Vec<u8>, Error>, Option<AllocationStats>) {
-		let Some(method_index) = self.0.module().lookup_export(name) else {
-			return (
-				Err(format!("cannot call into the runtime: export not found: '{name}'").into()),
-				None,
-			);
-		};
-
 		let Ok(raw_data_length) = u32::try_from(raw_data.len()) else {
 			return (
 				Err(format!("cannot call runtime method '{name}': input payload is too big").into()),
 				None,
 			);
 		};
 
-		// TODO: This will leak guest memory; find a better solution.
-		let mut state_args = polkavm::StateArgs::new();
-
 		// Make sure the memory is cleared...
-		state_args.reset_memory(true);
-		// ...and allocate space for the input payload.
-		state_args.sbrk(raw_data_length);
+		self.0.reset_memory().unwrap();
 
-		match self.0.update_state(state_args) {
-			Ok(()) => {},
-			Err(polkavm::ExecutionError::Trap(trap)) => {
-				return (Err(format!("call into the runtime method '{name}' failed: failed to prepare the guest's memory: {trap}").into()), None);
-			},
-			Err(polkavm::ExecutionError::Error(error)) => {
-				return (Err(format!("call into the runtime method '{name}' failed: failed to prepare the guest's memory: {error}").into()), None);
-			},
-			Err(polkavm::ExecutionError::OutOfGas) => unreachable!("gas metering is never enabled"),
-		}
+		// ...and allocate space for the input payload.
+		self.0.sbrk(raw_data_length).unwrap();
```

Comment:
Remove unwraps, and restore the comment about leaking memory.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/6533/commits/5b375075ad76fc7a0cefc6b6212b487a6682d73c